### PR TITLE
Fix leads priority values

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

File: src/redux/selectors/domainData.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
 // import { createSelector } from 'reselect';
 
 // NOTE: Use these to make sure reference don't change
-const emptyObject = {};
+const emptyInputValues = {
+    authors: [],
+    priority: 100,
```
Comment (tnagorra) [inline src/redux/selectors/domainData.js]: Same thing on the priority here. It should be set from leadOptions

Comment (AdityaKhatri) [reply src/redux/selectors/domainData.js]: Fixed

File: src/views/AddLead/index.js [hunk 2]
```diff
@@ -246,6 +234,13 @@ class AddLead extends React.PureComponent {
         }
     }
 
+    getPriorityOptions = memoize((priority, isProjectSelected) => {
+        if (!isProjectSelected || isNotDefined(priority)) {
+            return undefined;
+        }
+        return priority.sort((a, b) => compareNumber(a.key, b.key));
```
Comment (tnagorra) [inline src/views/AddLead/index.js]: We should make a copy before sorting a list

File: src/views/AddLead/utils.js [hunk 3]
```diff
@@ -51,6 +52,14 @@ export function fillExtraInfo(inputValues, currentUserId, leadOptions = {}) {
     ) {
         values.confidentiality = leadOptions.confidentiality[0].key;
     }
+    if (
+        !values.priority
```
Comment (tnagorra) [inline src/views/AddLead/utils.js:56]: why not use `isDefined` here? it's unsafe to use this syntax to check for undefined vaues for number.

File: src/views/AddLead/utils.js [hunk 4]
```diff
@@ -51,6 +52,14 @@ export function fillExtraInfo(inputValues, currentUserId, leadOptions = {}) {
...
+        && leadOptions.priority
+        && leadOptions.priority.length > 0
+    ) {
+        const sortedPriority = leadOptions.priority.sort((a, b) => compareNumber(a.key, b.key));
```
Comment (tnagorra) [inline src/views/AddLead/utils.js:60]: We should make a copy before we sort a list

**Review by tnagorra — CHANGES_REQUESTED** [review]
Let's fix these and we can deploy it.

**Pull request merged**